apps/workers/db-check/worker.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math"
+	"time"
+
+	"github.com/jbsmith7741/uri"
+	"github.com/pcelvng/task"
+	"github.com/pcelvng/task-tools/slack"
+)
+
+type worker struct {
+	options
+	task.Meta
+	slack *slack.Slack
+
+	Table     string    `uri:"table" required:"true"`      // name of the table to query
+	DateField string    `uri:"date_field" required:"true"` // the date field to get row counts
+	Date1     time.Time `uri:"date1" format:"2006-01-02"`  // default is 24h back from today
+	Date2     time.Time `uri:"date2" format:"2006-01-02"`  // 2nd date to compare records against date1
+	Offset    string    `uri:"offset"`                     // duration back from date1 used to determine the value of date2 if needed
+	Percent   float64   `uri:"percent" default:"0.05"`     // allowed difference between date1 and date2 row counts
+}
+
+func (o *options) newWorker(info string) task.Worker {
+	w := &worker{
+		options: *o, // copy option values
+		Meta:    task.NewMeta(),
+		Date1:   time.Now().Add(-time.Hour * 24), // default date is yesterday
+		slack:   &slack.Slack{Url: o.Slack},
+	}
+
+	if err := uri.Unmarshal(info, w); err != nil {
+		return task.InvalidWorker("params uri.unmarshal: %v", err)
+	}
+
+	return w
+}
+
+// RecordCount returns the row count for the given column field, and date
+func (w *worker) RecordCount(ctx context.Context, field string, date time.Time) (rows int, err error) {
+	q := "select count(0) r from " + w.Table + " where date(" + field + ") = $1"
```

Comment:
should we be using Prepared statements and Query Params instead of string construction?

Reply:
I've personally run into problems with prepare statements, but besides that I would have to create the prepare statement each time a task is sent. Since the column is dynamic it cannot be a static prepare, which seems to be the point for having prepare statements in the first place.